In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Exploratory Data Analysis:

## Viewing the Dataset:
First lets view the basic properties of our dataset by loading it and viewing it in the DataFrame format

In [2]:
#Loading the dataset
data_df = pd.read_csv("../Datasets/SampleSuperstore.csv")

In [3]:
#Viewing the first 5 rows of the dataset
data_df.head()

Ship Mode    Segment        Country             City       State  \
0    Second Class   Consumer  United States        Henderson    Kentucky   
1    Second Class   Consumer  United States        Henderson    Kentucky   
2    Second Class  Corporate  United States      Los Angeles  California   
3  Standard Class   Consumer  United States  Fort Lauderdale     Florida   
4  Standard Class   Consumer  United States  Fort Lauderdale     Florida   

   Postal Code Region         Category Sub-Category     Sales  Quantity  \
0        42420  South        Furniture    Bookcases  261.9600         2   
1        42420  South        Furniture       Chairs  731.9400         3   
2        90036   West  Office Supplies       Labels   14.6200         2   
3        33311  South        Furniture       Tables  957.5775         5   
4        33311  South  Office Supplies      Storage   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164

In [9]:
#Viewing the information for the dataset
data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9994 non-null   object 
 1   Segment       9994 non-null   object 
 2   Country       9994 non-null   object 
 3   City          9994 non-null   object 
 4   State         9994 non-null   object 
 5   Postal Code   9994 non-null   int64  
 6   Region        9994 non-null   object 
 7   Category      9994 non-null   object 
 8   Sub-Category  9994 non-null   object 
 9   Sales         9994 non-null   float64
 10  Quantity      9994 non-null   int64  
 11  Discount      9994 non-null   float64
 12  Profit        9994 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1015.1+ KB


The above information indicates that there are 9994 entries out of which none of the attributes carries missing values, thus, we do not have to perform operations on missing values. Although we have alot of object datatypes indicating that there are a mixed type of values in that attribute, but all such columns refer to String based attributes, so they won't cause a problem.

## Finding Relationships Between Attributes:
Now that we have seen our data, we need to see which attributes have relationships and dependencies on others. 